src/v/raft/consensus.cc

```diff
@@ -1384,6 +1385,7 @@ consensus::disk_append(model::record_batch_reader&& reader) {
              _log.make_appender(cfg),
              cfg.timeout)
       .then([this](std::tuple<ret_t, std::vector<offset_configuration>> t) {
+          _disk_append.broadcast();
```

Comment:
is this occurring _after_ the operations that advance the dirty offset have completed? (i can't remember what all `for_each_ref_extract_configuration` does).

Reply:
This `_disk_append.broadcast()` occur after data were successfully appended to the log and dirty offset was advanced. It is after the reader was consumed by the log appender.